### ISSUE TITLE
Longterm retention of data about status changes

### ISSUE THREAD

**Issue by hasso** [guardian/alerta#235]
Due to operational requirements I need to preserve all the data about alerts and changes their status. There is a syslog plugin, but it doesn't cover my needs for various reasons.
- It's very noisy in case of using "send fast duplicates in case of alert situation, stop is resolved and let it timeout in alerta" alerts.
- It doesn't log internal status changes.

What's really needed is a hook to react to every change in status and write it down (syslog, file) with info about alert.

**Comment by satterly** [comment]
You can do this using the mongoDB [oplog](https://docs.mongodb.com/v3.2/core/replica-set-oplog/). See https://www.mongodb.com/presentations/building-real-time-systems-mongodb-using-oplog-stripe

**Comment by satterly** [comment]
For example, an `ack` looks like this...

```
{  
   "ts":Timestamp(1478158634,
   2   ),
   "t":NumberLong(1),
   "h":NumberLong("-8642367183747826398"),
   "v":2,
   "op":"u",
   "ns":"monitoring.alerts",
   "o2":{  
      "_id":"8cecf952-c2d5-4488-ba11-2de1fd61772a"
   },
   "o":{  
      "$set":{  
         "status":"ack",
         "history":[  
            {  
               "updateTime":               ISODate("2016-11-03T00:49:42.954               Z"),
               "severity":"normal",
               "text":"",
               "value":"n/a",
               "event":"e1",
               "type":"external",
               "id":"8cecf952-c2d5-4488-ba11-2de1fd61772a"
            },
            {  
               "status":"closed",
               "updateTime":               ISODate("2016-11-03T00:49:43.021               Z"),
               "text":"new alert status change",
               "id":"8cecf952-c2d5-4488-ba11-2de1fd61772a",
               "type":"auto",
               "event":"e1"
            },
            {  
               "updateTime":               ISODate("2016-11-03T07:35:54.363               Z"),
               "severity":"warning",
               "text":"",
               "value":"n/a",
               "event":"e1",
               "type":"external",
               "id":"74003bed-96d6-48dd-97f5-21214a453e08"
            },
            {  
               "status":"open",
               "updateTime":               ISODate("2016-11-03T07:35:54.363               Z"),
               "text":"correlated alert status change",
               "id":"74003bed-96d6-48dd-97f5-21214a453e08",
               "type":"auto",
               "event":"e1"
            },
            {  
               "status":"ack",
               "updateTime":               ISODate("2016-11-03T07:37:14.798               Z"),
               "text":"",
               "id":"8cecf952-c2d5-4488-ba11-2de1fd61772a",
               "type":"external",
               "event":"e1"
            }
         ]
      }
   }
}
```

**Comment by satterly** [comment]
I'm going to leave this as an exercise for the user. Let me know if you need any more info.

**Comment by satterly** [comment]
Fixed by #744
